### PR TITLE
feat(frontend): new modal design

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

New design for the modal as in Figma.

# TODO

We need E2E tests that take screenshots of the modals. It was incredibly painful to test each modal repeatedly. I ended up taking screenshots for this PR about 20 times.

# Leftovers

It was already too much work. I'll have a look to the WalletConnect modal afterwards.

# Changes

- Adapt style for the gix-cmp modals
- Create a mixin for the card-content of the modal because we need a special tricks on the "Manage tokens" modal
- Adapt various paddings
- Modify "Manage Tokens" to fit this new design
- Remove custom style in "Dapp modal"

# Screenshots

Capture d’écran 2024-10-29 à 09 42 13
Capture d’écran 2024-10-29 à 09 42 23
Capture d’écran 2024-10-29 à 09 42 25
Capture d’écran 2024-10-29 à 09 42 32
Capture d’écran 2024-10-29 à 09 43 03
Capture d’écran 2024-10-29 à 09 43 25
Capture d’écran 2024-10-29 à 09 43 27
Capture d’écran 2024-10-29 à 09 43 33
Capture d’écran 2024-10-29 à 09 43 42
Capture d’écran 2024-10-29 à 09 43 46
Capture d’écran 2024-10-29 à 09 43 49
Capture d’écran 2024-10-29 à 09 43 55
Capture d’écran 2024-10-29 à 09 43 57
Capture d’écran 2024-10-29 à 09 44 03
Capture d’écran 2024-10-29 à 09 44 07
Capture d’écran 2024-10-29 à 09 44 16
Capture d’écran 2024-10-29 à 09 44 23
Capture d’écran 2024-10-29 à 09 44 33
Capture d’écran 2024-10-29 à 09 44 45
Capture d’écran 2024-10-29 à 09 44 48
Capture d’écran 2024-10-29 à 09 44 51
Capture d’écran 2024-10-29 à 09 44 54
Capture d’écran 2024-10-29 à 09 44 55
Capture d’écran 2024-10-29 à 09 44 59
Capture d’écran 2024-10-29 à 09 45 01
Capture d’écran 2024-10-29 à 09 45 03
Capture d’écran 2024-10-29 à 09 45 10
Capture d’écran 2024-10-29 à 09 45 12
Capture d’écran 2024-10-29 à 09 45 17
Capture d’écran 2024-10-29 à 09 45 19
Capture d’écran 2024-10-29 à 09 45 23
Capture d’écran 2024-10-29 à 09 45 25
Capture d’écran 2024-10-29 à 09 45 27

### CODE REVIEW

File: src/frontend/src/lib/components/manage/ManageTokens.svelte [hunk 1]
```diff
@@ -235,37 +234,39 @@
...
-	<Hr />
-
 	<button
-		class="flex w-full justify-center pb-5 pt-4 text-center font-bold text-blue-ribbon no-underline"
+		class="mb-4 flex w-full justify-center pt-4 text-center font-bold text-blue-ribbon no-underline"
 		on:click={() => dispatch('icAddToken')}>+ {$i18n.tokens.manage.text.do_not_see_import}</button
 	>
+{/if}
 
+{#if !noTokensMatch}
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/manage/ManageTokens.svelte:269]: out of curiosity: why move it to its own IF condition?

File: src/frontend/src/lib/styles/global/gix.scss [hunk 2]
```diff
@@ -115,9 +116,9 @@ div.input-field input[id]::placeholder {
 }
 
 div.modal {
-	--overlay-background: var(--color-white);
+	--overlay-background: #f8f9fa;
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/styles/global/gix.scss:119]: I think you created `ghost-white` for this?

File: src/frontend/src/lib/styles/global/gix.scss [hunk 3]
```diff
@@ -115,9 +116,9 @@ div.input-field input[id]::placeholder {
...
 	--overlay-background-contrast: var(--color-secondary);
-	--overlay-content-background: var(--color-white);
+	--overlay-content-background: #f8f9fa;
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/styles/global/gix.scss:121]: ditto

File: src/frontend/src/lib/styles/global/gix.scss [hunk 4]
```diff
@@ -171,9 +175,22 @@ div.modal {
...
+			min-height: auto;
+		}
+
 		> div.stretch {
 			flex: 1;
-			margin: 0 0 var(--padding-3x);
+
+			margin: 0 0 var(--padding-0_5x);
+
+			&:has(+ *) {
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/styles/global/gix.scss:187]: I am not sure, but doesn't the new version of `SASS` says that this must be after the other "direct" declarations? Or it is not flagged for `svelte` files?

**Comment by AntonioVentilii-DFINITY** [comment]
@peterpeterparker completly agree on the E2E tests + mobile screenshots. I am tagging @stefanberger-difnity

File: src/frontend/src/lib/styles/global/gix.scss (hunk 2 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/styles/global/gix.scss]: Ah yeah good catch thanks. Totally forgot about it, I was so focus having to redo these freaking screenshots zillion times. Changed.

File: src/frontend/src/lib/components/manage/ManageTokens.svelte (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/components/manage/ManageTokens.svelte]: Good catch, thanks! That's a leftover of one of the 20th iterations. I removed the additional if. Code similar as main.

File: src/frontend/src/lib/styles/global/gix.scss (hunk 4 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/styles/global/gix.scss:191]: No warnings 🤷‍♂️. I moved it anyway, you never know.

**Pull request merged**